basher.go

```diff
@@ -232,10 +233,22 @@ func (c *Context) Run(command string, args []string) (int, error) {
 	for _, arg := range args {
 		argstring = argstring + " '" + arg + "'"
 	}
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals)
+
 	cmd := exec.Command(c.BashPath, "-c", command+argstring)
 	cmd.Env = []string{"BASH_ENV=" + envfile}
 	cmd.Stdin = c.Stdin
 	cmd.Stdout = c.Stdout
 	cmd.Stderr = c.Stderr
-	return exitStatus(cmd.Run())
+	cmd.Start()
+	go func() {
+		for sig := range signals {
+			if !cmd.ProcessState.Exited() {
```

Comment:
I can't tell from this trace for some reason, but that usually means something is `nil` you were expecting not to be. Probably `cmd.ProcessState` if that's not created until the process starts (which makes sense). So you'd want to do `if cmd.ProcessState != nil` ...